### ISSUE TITLE
Invalid PDF structure

### ISSUE THREAD

**Issue by thnk2wn** [mozilla/pdf.js#5742]
With some PDFs I get the below error:

PDF.js v1.0.907 (build: e9072ac)
Message: Invalid PDF structure

I can open the PDF fine in Acrobat Reader. Not sure if this is it or not but this happens to be an EchoSigned PDF document.

For now an example file to reproduce is here: https://www.dropbox.com/s/dkpspp3llvqp7o9/52635.pdf?dl=0

**Comment by thnk2wn** [comment]
So I tried again a couple times and now it works. It seems to be sporadic. 

If it matters I'm using Google Chrome (40.0.2214.111 m). We had similar "cannot load PDF" issues before using iFrame or <object> style with Chrome and switched to PdfJs to help avoid this issue but maybe it's more of a Chrome problem.

**Comment by yurydelendik** [comment]
Does it work for other browsers? If yes, can be #5512. Also v1.0.907 is older version of PDF.js.

**Comment by thnk2wn** [comment]
It seems to just be Chrome. Even there it is sporadic and it can happen randomly hitting refresh.

I took a look at the related issue you tagged but can't quite tell what, if anything, I can do to work around it yet.

As to the version, 1.0.907 was the latest stable version that I just downloaded today from http://mozilla.github.io/pdf.js/getting_started/#download. But if the beta is stable and might help with this issue I'm open to switching to that version.

**Comment by CodingFabian** [comment]
@thnk2wn for our product we take the latest version of pdf.js once we want to release and run it through our automated tests. Most of the time it is much better than the latest "official stable build". That one is more a marker for the firefox inclusion. IMHO the current master as as stable as any release, if not better.
In general bug reports will be only accepted on the latest version to avoid hunting already fixed bugs.

**Comment by CodingFabian** [comment]
and yes from your description it sounds likely to be #5512. The newer versions work around this issue and hopefully Chrome decides soon how they want to fix it :-)

**Comment by Snuffleupagus** [comment]
@thnk2wn Please re-test this using the *latest* version of PDF.js, which as of this writing is `1.0.1147`.

You can test this easily using the demo viewer, see http://mozilla.github.io/pdf.js/web/viewer.html; or if you prefer to test locally you can use the archive from the `gh-pages` branch, see https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#gh-pages.

**Comment by thnk2wn** [comment]
@yurydelendik, @CodingFabian, @Snuffleupagus - Thanks guys. So far I cannot reproduce the issue using 1.0.1147 but I'm waiting on more thorough testing from QA to be sure.

Time permitting I might look into creating a http://www.nuget.org/ package for this to make installing and upgrading Pdf.Js easier for the .net / asp.net world (didn't see anything out there).

Also looking for a way to display a loading indicator while the viewer loads since it's slower than native object tag but these are separate topics.

**Comment by thnk2wn** [comment]
Looks good thanks

**Comment by ananyaojha** [comment]
Invalid or corrupted file.
PDF.js v1.1.114 (build: 3fd44fd)
Message: Invalid PDF structure

It's coming in all browsers.
![screencapture-192-168-1-17-wmt_go_livedb_mig11-files-1454758363986](https://cloud.githubusercontent.com/assets/5165685/12866440/7b72da76-ccf3-11e5-979f-1fbc56367f28.png)

**Comment by m-hcb** [comment]
I have the same issue as @ananyaojha. any solution guys? or reason?

**Comment by timvandermeij** [comment]
Use the most recent PDF.js version. Otherwise open a new issue with all requested details provided, such as an example PDF file to reproduce the issue.

**Comment by tofanelli** [comment]
I'm getting this same error message with the PDF file provided with the PDF.js package (in the picture i just renamed it for the sake of tests).

![image](https://user-images.githubusercontent.com/5512922/41124900-be502fbc-6a79-11e8-9d0c-d8ae0148e03e.png)

Chrome Version 66.0.3359.181 (Official Build) (64-bit)
and
Firefox 60.0.1 (64-bit)
Windows 8.1

Any lights?

**Comment by bloigge** [comment]
Same for me. Running pdf.js in ionic cordova on android (chrome web view)

**Comment by Edward-Shaw** [comment]
Meet the same problem with cordova on android.

> Same for me. Running pdf.js in ionic cordova on android (chrome web view)

**Comment by tofanelli** [comment]
guys... I solved my issued using a HTML native tag

`<object width="400" height="400" data="myFile.pdf" type="application/pdf"></object>`

and done... the browser itself does the job to understand the object and open it using its tools... way better than add more things (libs) to your project =)

**Comment by darkomenx** [comment]
Same issue with latest version of pdfjs at now (2.5.207) into Electron application and this code : 

`<iframe height="100%" width="100%" :src="'pdfjs/web/viewer.html?file='+path"></iframe>`

path is created like this : `this.path = encodeURIComponent('/Users/xxxx/xxxx/test/dummy.pdf')`

and with this code nothing happens : 

`<object width="400" height="400" data="myFile.pdf" type="application/pdf"></object>`

- my test file pdf (dummy.pdf) opened correctly with external viewer pdf
- if I manually open pdf into my iframe viewer for pdfjs the pdf is correctly viewed

**Comment by axanthus** [comment]
I was having trouble displaying a simple pdf in the browser, but I figured out it was my url file path. After fetching the binary data from the DB, I converted into a base64 buffer stream, then converted it into a blob with type pdf, and then created an Object URL based on the blob. 

 ` const arrayView = Buffer.from(sclass.syllabus.binData, 'base64'`
  `const blob = new Blob([arrayView], { type: 'application/pdf' })`
 ` const url = window.URL.createObjectURL(blob)`
  `sclass.syllabus.binData = url;`

Using that URL, I was able to display my pdf in both an iframe and object element:

`<iframe id="pdf-js-viewer" title="webviewer" frameBorder="0" width="100%" height="100%"`
     ` src={'/lib/web/viewer.html?file=${sclass.syllabus.binData}'}`
` > `
` </iframe>`

Or:

`<object`
`    data={sclass.syllabus.binData}`
`     type='application/pdf'`
`     width='100%'`
`     height='100%'`
`      aria-label='PDF Document'`
/>
